opensearch_py_ml/ml_commons/validators/model_access_control.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+""" Module for validating model access control parameters """
+
+from typing import List, Optional
+
+ACCESS_MODES = ["public", "private", "restricted"]
+
+NoneType = type(None)
+
+
+def _validate_model_group_name(name: str):
+    if not name or not isinstance(name, str):
+        raise ValueError("name is required and needs to be a string")
+
+
+def _validate_model_group_description(description: Optional[str]):
+    if not isinstance(description, (NoneType, str)):
+        raise ValueError("description needs to be a string")
+
+
+def _validate_model_group_access_mode(access_mode: Optional[str]):
+    if access_mode is None:
+        return
+    if access_mode not in ACCESS_MODES:
+        raise ValueError(f"access_mode can must be in {ACCESS_MODES} or None")
+
+
+def _validate_model_group_backend_roles(backend_roles: Optional[List[str]]):
+    if not isinstance(backend_roles, (NoneType, list)):
+        raise ValueError("backend_roles should either be None or a list of roles names")
+
+
+def _validate_model_group_add_all_backend_roles(add_all_backend_roles: Optional[bool]):
+    if not isinstance(add_all_backend_roles, (NoneType, bool)):
+        raise ValueError("add_all_backend_roles should be a boolean")
+
+
+def _validate_model_group_query(query: dict, operation: Optional[str] = None):
+    if not isinstance(query, dict):
+        raise ValueError("query needs to be a dictionary")
+
+    if operation and not isinstance(operation, str):
+        raise ValueError("operation needs to be a string")
+
+
+def validate_create_model_group_parameters(
```

Comment:
Any specific reason we are not validating all the exception cases while creating model group? Like the ones we have here: https://github.com/opensearch-project/ml-commons/blob/6efb1ebeefb8a7930f6767262457d0d5b568ac75/plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java#L148

Reply:
@rbhavna , since, these validations are already being done by ml-commons, having them here as well felt like double validations.

Reply:
Just curious, the three exception cases that we are testing [here](https://github.com/opensearch-project/opensearch-py-ml/pull/332/files#diff-7b43ebfd8a955d722cd6f036c88ae4f3d69bde277138a85aa9d6654739cd3ee8R65) are also being validated by ml-commons. Any reason we still have them here and not the others?

Reply:
Right. I do feel we shouldn't be having these checks in `opensearch-py-ml`. Because, if checks on `ml-commons` change, we again need to update this on the client as well.  I was thinking, client libraries only purpose would be to make the calls and if any errors from the `opensearch`, it should be able to pass them to user without any change. Also, maybe add few utility functions to make things easier for the user.

@dhrubo-os , @rbhavna , what do you think about removing these validations form `opensearch-py-ml`? Or Should I go ahead with implementing the validations from `ml-commons` ?

Reply:
As `opensearch-py-ml` is a client plugin for ml-commons, we should have only the client side testing similar like any web application. Some basic input validations which can be easily caught before reaching to the server. We don't need to add that type of validations for which we need to make a back-end call like testing if the user is admin or not. 

Yeah we need to have the same basic validations in ml-commons too, as py-ml isn't the only client for ml-commons.

Reply:
+1